pkg/errors/errors.go

```diff
@@ -8,59 +8,170 @@ import (
+	UpdateFailed                error = NewMocError(moccodes.UpdateFailed)
+	NotInitialized              error = NewMocError(moccodes.NotInitialized)
+	NotImplemented              error = NewMocError(moccodes.NotImplemented)
+	OutOfRange                  error = NewMocError(moccodes.OutOfRange)
+	AlreadySet                  error = NewMocError(moccodes.AlreadySet)
+	NotSet                      error = NewMocError(moccodes.NotSet)
+	InconsistentState           error = NewMocError(moccodes.InconsistentState)
+	PendingState                error = NewMocError(moccodes.PendingState)
+	WrongHost                   error = NewMocError(moccodes.WrongHost)
+	PoolFull                    error = NewMocError(moccodes.PoolFull)
+	NoActionTaken               error = NewMocError(moccodes.NoActionTaken)
+	Expired                     error = NewMocError(moccodes.Expired)
+	Revoked                     error = NewMocError(moccodes.Revoked)
+	Timeout                     error = NewMocError(moccodes.Timeout)
+	RunCommandFailed            error = NewMocError(moccodes.RunCommandFailed)
+	InvalidToken                error = NewMocError(moccodes.InvalidToken)
+	Unknown                     error = NewMocError(moccodes.Unknown)
+	DeleteFailed                error = NewMocError(moccodes.DeleteFailed)
+	DeletePending               error = NewMocError(moccodes.DeletePending)
+	FileNotFound                error = NewMocError(moccodes.FileNotFound)
+	PathNotFound                error = NewMocError(moccodes.PathNotFound)
+	NotEnoughSpace              error = NewMocError(moccodes.NotEnoughSpace)
+	AccessDenied                error = NewMocError(moccodes.AccessDenied)
+	BlobNotFound                error = NewMocError(moccodes.BlobNotFound)
+	GenericFailure              error = NewMocError(moccodes.GenericFailure)
+	NoAuthenticationInformation error = NewMocError(moccodes.NoAuthenticationInformation)
+	MeasurementUnitError        error = NewMocError(moccodes.MeasurementUnitError)
+	QuotaViolation              error = NewMocError(moccodes.QuotaViolation)
+	IPOutOfRange                error = NewMocError(moccodes.IPOutOfRange)
+	MultipleErrors              error = NewMocError(moccodes.MultipleErrors)
 )
 
+// GetMocErrorCode attempts to extract the MocCode from the given error. If the error is a multierr,
+// it will return the MocCode if all errors in the multierr match the same MocCode. If there are
+// multiple non-matching MocCodes, then it returns MocCode.MultipleErrors and false.
+//
+// GetMocErrorCode follows the following rules when parsing individual errors:
+//
+//   - If the error is nil, it returns MocCode.OK and true.
+//   - If the error is of type MocError, it returns the MocCode and true.
+//   - If the error has a Cause that is not nil and is of type MocError, it returns the MocCode of the Cause and true.
+//   - If both the error and its Cause are not of type MocError, it returns MocCode.Unknown and false.
+func GetMocErrorCode(err error) (moccodes.MocCode, bool) {
```

Comment:
Nit: for my own understanding, what is the intent of the boolean?

Reply:
Looks like it's used below in checkErrors but since one arg will always be a valid MOC error, isn't the equality check enough?

Reply:
It essentially is just a "was the error _actually_ Unknown or MultipleErrors? Or were we just unable to interpret the code / the error has multiple different codes?"

Reply:
Closing to help me organize PR revisions, feel free to re-open if I didn't address something.